actors/runtime/runtime.go

```diff
@@ -123,8 +124,49 @@ type TraceSpan interface {
 	End()
 }
 
-type ActorStateHandle interface {
-	UpdateRelease(newStateCID abi.ActorSubstateCID)
-	Release(checkStateCID abi.ActorSubstateCID)
-	Take() abi.ActorSubstateCID
-}
\ No newline at end of file
+// ReadonlyStateHandle provides read-only access to an actor's state object.
+type ReadonlyStateHandle interface {
+	// Readonly loads a readonly copy of the state into the argument.
+	//
+	// Any modification to the state is illegal and will result in an abort.
+	Readonly(obj CBORMarshalable)
```

Comment:
There's no closure or release here because nothing can detect or prevent subsequent inspection of the state object in any case.